### ISSUE TITLE
S3 backend: terraform_remote_state seems broken on v1.6.0-beta1

### ISSUE THREAD

**Issue by bryceml** [hashicorp/terraform#33884]
### Terraform Version

```shell
Terraform v1.6.0-dev
on darwin_arm64
+ provider registry.terraform.io/hashicorp/archive v2.4.0
+ provider registry.terraform.io/hashicorp/aws v5.16.2
+ provider registry.terraform.io/hashicorp/external v2.3.1
```

### Terraform Configuration Files

```terraform
data "terraform_remote_state" "read_only" {
  backend = "s3"

  config = {
    bucket  = local.s3_backend_bucket
    key     = "aws/states/read_only/terraform.tfstate"
    region  = local.primary_region
    profile = local.profile
  }
}
```

### Debug Output

https://gist.github.com/bryceml/3dd98ea69d1a1e4a78e2c168c9a3721f

### Expected Behavior

It should have not errored telling me `assume_role` and `endpoints` are required, because they are not.

### Actual Behavior

```
│ Error: Invalid backend configuration
│ 
│   with module.ue1.data.terraform_remote_state.read_only,
│   on regional/main.tf line 16, in data "terraform_remote_state" "read_only":
│   16:   config = {
│   17:     bucket  = local.s3_backend_bucket
│   18:     key     = "aws/states/read_only/terraform.tfstate"
│   19:     region  = local.primary_region
│   20:     profile = local.profile
│   21:   }
│ 
│ The given configuration is not valid for backend "s3": attribute
│ "assume_role" is required.
╵
╷
│ Error: Invalid backend configuration
│ 
│   with module.uw2.data.terraform_remote_state.read_only,
│   on regional/main.tf line 16, in data "terraform_remote_state" "read_only":
│   16:   config = {
│   17:     bucket  = local.s3_backend_bucket
│   18:     key     = "aws/states/read_only/terraform.tfstate"
│   19:     region  = local.primary_region
│   20:     profile = local.profile
│   21:   }
│ 
│ The given configuration is not valid for backend "s3": attribute
│ "endpoints" is required.
```

### Steps to Reproduce

terraform init
terraform plan

**Comment by crw** [comment]
Thanks for this report!

**Comment by jar-b** [comment]
Thanks again for the report, @bryceml! This has been fixed and backported to 1.6. You should be able to use this data source again in the next 1.6 pre-release.
